app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt

```diff
@@ -200,41 +232,65 @@ class SellTransactionDetailsBottomSheet :
         val labelAmount: String
+        val receiverAddress: String
+        val receiverTitle: String = getString(
+            if (details.status == SellTransactionStatus.PENDING) {
+                R.string.main_will_be_send_to
+            } else {
+                R.string.main_send_to
+            }
+        )
         when (details.status) {
             SellTransactionStatus.WAITING_FOR_DEPOSIT -> {
                 boldAmount = getString(
                     R.string.sell_lock_token_amount, tokenAmount, Constants.SOL_SYMBOL
                 )
-                labelAmount = getString(
-                    R.string.sell_lock_waiting_for_deposit_fiat_amount, fiatAmount, fiatAbbreviation
+                labelAmount = emptyString()
+                receiverAddress = details.receiverAddress.let {
+                    if (details.isReceiverAddressWallet) it.cutMiddle() else it
+                }
+            }
+            SellTransactionStatus.PENDING -> {
+                boldAmount = getString(
+                    R.string.sell_lock_pending_fiat_amount, tokenAmount, Constants.SOL_SYMBOL,
                 )
+                labelAmount = emptyString()
+                receiverAddress = getString(R.string.sell_details_receiver_moonpay_bank)
             }
-            SellTransactionStatus.PENDING, SellTransactionStatus.COMPLETED -> {
+
+            SellTransactionStatus.COMPLETED -> {
                 boldAmount = getString(
-                    R.string.sell_lock_pending_fiat_amount, fiatAmount, fiatAbbreviation
+                    R.string.sell_lock_token_amount, fiatAmount, fiatAbbreviation,
                 )
                 labelAmount = getString(
                     R.string.sell_lock_token_amount, tokenAmount, Constants.SOL_SYMBOL
                 )
+                receiverAddress = getString(R.string.sell_details_receiver_moonpay_bank)
             }
             SellTransactionStatus.FAILED -> {
                 boldAmount = getString(
                     R.string.sell_lock_token_amount, tokenAmount, Constants.SOL_SYMBOL
                 )
                 labelAmount = emptyString()
+                receiverAddress = emptyString()
             }
         }
         textViewAmount.text = boldAmount
         textViewFiatValue.text = labelAmount
+        textViewFiatValue.isVisible = labelAmount.isNotEmpty()
+        textViewReceiverTitle.text = receiverTitle
 
         containerReceiver.isVisible = details.status != SellTransactionStatus.FAILED
-        textViewReceiverAddress.text = details.receiverAddress.let {
-            if (details.isReceiverAddressWallet) it.cutMiddle() else it
-        }
+        textViewReceiverAddress.text = receiverAddress
     }
 
     private fun renderCopyButton() = with(binding.layoutDetails.imageViewCopy) {
-        isVisible = details.isReceiverAddressWallet
+        isVisible = when (details.status) {
+            SellTransactionStatus.WAITING_FOR_DEPOSIT -> details.isReceiverAddressWallet
```

Comment:
it doesn't work without this check? i thought this flag would be used to hide copy image and we set this flag outside of bottom sheet

Reply:
no, on other status not need to show receiver address

Reply:
receiverAddress need to show only when WAITING_FOR_DEPOSIT status